sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -631,6 +631,33 @@ object SQLConf {
     .intConf
     .createWithDefault(200)
 
+  val SQLSTREAM_WATERMARK_ENABLE = buildConf("spark.sqlstreaming.watermark.enable")
+    .doc("Whether use watermark in sqlstreaming.")
+    .booleanConf
+    .createWithDefault(false)
+
+  val SQLSTREAM_OUTPUTMODE = buildConf("spark.sqlstreaming.outputMode")
+    .doc("The output mode used in sqlstreaming")
+    .stringConf
+    .createWithDefault("append")
+
+  val SQLSTREAM_TRIGGER = buildConf("spark.sqlstreaming.trigger")
```

Comment:
we have so many configurations, i think in thrift server scenarios where user can open multiple sessions and run streaming query based on different query context. each query will be requiring its own context of trigger intervals,water marking,windowing. can you elaborate a bit how we address these scenarios.

Reply:
I don't think there are any problems with this. SQLStreaming is using Command to run streaming query, which is similar to InsertIntoHiveTable. herefore, the batch SQL and streaming SQL solution is expected.In addition, currently an application can only run one streaming SQL.
Therefore, the batch SQL and streaming SQL solution is expected

Reply:
so here stream-stream join is not supported right?  to elaborate can i create two stream source tables and then join both and write to sink?
because if i want to create two streams for 2 different topics, i may need to provide different configurations for watermark or window or rigger interval.

Reply:
> insert into kafka_sql_out select stream t1.value from (select cast(value as string), timestamp as time1 from kafka_sql_in1) as t1 inner join (select cast(value as string), timestamp as time2 from kafka_sql_in2) as t2 on time1 >= time2 and time1 <= time2 + interval 10 seconds where t1.value == t2.value

No, SQLStreaming support stream join stream. The watermark config is put in the table properties.
As for trigger interval, different sources in stream join stream scene needs different trigger config?

Reply:
I mean the configurations like Triggers/outputmodes shall be configured within the scope of a particular stream context, currently its scope is  application level. this will have certain limitation like
eg: if i want to read the stream from multiple topics and write to sink after joining  the data from multiple topics . i think its not possible i guess. correct me if i miss something.

Reply:
> if i want to read the stream from multiple topics and write to sink after joining the data from multiple topics . 

Read from different topics has already supported in SQLStreaming.

> I mean the configurations like Triggers/outputmodes shall be configured within the scope of a particular stream context, currently its scope is application level.

Trigger and outputModes are the configurations for DataStreamWriter, not for DataStream Reader. I think you are trying to  run multi streaming in the same application and output to multi Sinks. Am I right?

Reply:
For now, I have not find a way to run multi query in the same time. Perhaps, we should write the configuration of sink into table? So that, we can support different sink in SQLStreaming. Do you agree with this?
Or, do you have any better idea?

Reply:
As i mentioned a  scenario above i am having few more suggestions and  thought about the solution for tackling these problems, this includes to provide sql CREATE STREAM to define a stream job with its own stream properties defined , also DDL's STOP STREAM and SHOW STREAMS TO show the status of the current running streams on a particular table which Wehnchen has already mentioned in mail. i will be able to share you a draft document by Monday. you can have a look and if you feel its fine  i will be very happy to contribute this as part of this feature.

Reply:
I had send the document in mail, please have a look. let me know for any clarifications.